api/output_policy.go

```diff
@@ -4,18 +4,22 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+	"strconv"
 	"strings"
 )
 
 const (
 	ErrOutputPolicyRequest = "output a policy, please"
+
+	listKey = "list"
```

Comment:
There are two schools of thought on this one...

Some people think all strings which are in some way a symbol should be factored out to constants.

Some people think that excessive factoring out of strings to constants actually obfuscates the code, because someone with reasonable familiarity of the Vault HTTP API will aready understand the significance of `"list"` as a query-parameter, but will need to double-check (or assume) what the local `listKey` constant actually has as a value to reach the same understanding.

I'm just an external contributor, so my opinion has no authority behind it whatsoever, but I am firmly in the second school of thought - inline it, don't people look elsewhere for a local constant for what is already a defined API.

But, possibly more significantly, the existing code for actually handling requests with this parameter, in Vault's `http/logical.go` is also written with `"list"` inlined, rather than as a separate constant.

Reply:
I think the last point is the most important to this particular change, and that this should be consistent with the way this is handled elsewhere in the code base. As such I updated it to inline it. I still personally believe that magic matching strings should be abstracted to constants and only ever referenced as such but it looks like that is not what we do in the vault code base.